project/src/main/webapp/closebrackets.js

```diff
@@ -0,0 +1,191 @@
+// CodeMirror, copyright (c) by Marijn Haverbeke and others
+// Distributed under an MIT license: https://codemirror.net/LICENSE
+
+(function(mod) {
```

Comment:
Have to say stuff in this file looks error-prone. Parsing is generally non-trivial. I am fine with merging this for now so we have a solution, but think about how this can be improved, and add more comments on what's going on throughout this file.